A##-list-health.md

```diff
@@ -0,0 +1,59 @@
+# Add List endpoint to gRPC Health service
+
+* **Author(s):** Marcos Huck
+* **Approver:** TBD
+* **Status:** Draft  
+* **Implemented in:** TBD
+* **Last updated:** 2024-12-17  
+* **Discussion at:** TBD
+
+
+## Abstract
+
+This proposal introduces a new `List` RPC endpoint for the Health service, allowing clients to retrieve the statuses of all monitored services or a filtered list of specific services. This feature simplifies integration with status-reporting dashboards and enhances observability for microservices.
+
+## Background
+
+The existing Health service provides basic health check functionality but lacks a mechanism to retrieve a comprehensive list of all monitored services and their statuses. 
+
+This limitation makes it challenging for clients to aggregate health information across multiple services, particularly for use cases like publishing service statuses to dashboards such as [Cachet](https://cachethq.io/) or [Statuspage](https://www.atlassian.com/software/statuspage).
```

Comment:
Can you help me understand the use-case here?  Both of those publishing services seem to be aimed at communicating overall service health to end-users.  However, the gRPC health service is really about exposing the health of an individual endpoint, not of the service as a whole.  In what context would these be used together?

Reply:
Thanks for the review. 

I imagine the following use case: 

We need to aggregate these endpoint statuses to get overall service health. 

We'd build an aggregator service that polls these endpoints, applies logic to determine service status, and then feeds that consolidated data to tools like `Cachet` or `Statuspage` for end-user visibility, and also for internal monitoring. So, while `gRPC` is endpoint-focused, the Health service is a key data source for a broader service health system. 

Adding a new `List` endpoint would allow towards that end.